From 8737969b9b79b946bf7cd77e88cc5ff0be21ff20 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:04:51 -0400
Subject: [tools] perf: Put proper period for for samples without PERIOD sample_type

Message-id: <1410181752-25631-47-git-send-email-jolsa@redhat.com>
Patchwork-id: 91717
O-Subject: [PATCH RHEL7.1 BZ1134356 046/307] perf tools: Put proper period for for samples without PERIOD sample_type
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit bc5290869d0a7f7abbde76ac95a7f7b6f5d7bb7b
Author: Jiri Olsa <jolsa@redhat.com>
Date: Mon Feb 3 12:44:41 2014 +0100

description
===========
We use PERF_SAMPLE_PERIOD sample type only for frequency
setup -F (default) option. The -c does not need store period,
because it's always the same.

In -c case the report code uses '1' as  period. Fixing
it to perf_event_attr::sample_period.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/evsel.c b/tools/perf/util/evsel.c
index 55407c5..c6f8ce9 100644
--- a/tools/perf/util/evsel.c
+++ b/tools/perf/util/evsel.c
@@ -1220,7 +1220,7 @@ int perf_evsel__parse_sample(struct perf_evsel *evsel, union perf_event *event,
  memset(data, 0, sizeof(*data));
  data->cpu = data->pid = data->tid = -1;
  data->stream_id = data->id = data->time = -1ULL;
- data->period = 1;
+ data->period = evsel->attr.sample_period;
  data->weight = 0;
 
  if (event->header.type != PERF_RECORD_SAMPLE) {
-- 
1.7.1